quasar-utils/src/main/groovy/net/corda/plugins/quasar/QuasarExtension.groovy

```diff
@@ -53,45 +58,51 @@ class QuasarExtension {
         ObjectFactory objects,
         String defaultGroup,
         String defaultVersion,
-        String defaultClassifier,
+        String defaultSuspendable,
         Iterable<? extends String> initialPackageExclusions,
         Iterable<? extends String> initialClassLoaderExclusions
     ) {
         group = objects.property(String).convention(defaultGroup)
         version = objects.property(String).convention(defaultVersion)
-        classifier = objects.property(String).convention(defaultClassifier)
-        dependency = group.flatMap { grp ->
-            version.flatMap { ver ->
-                classifier.map { cls ->
-                    [ group: grp, name: 'quasar-core', version: ver, classifier: cls, ext: 'jar' ]
-                }
-            }
+        dependency = group.zip(version) { grp, ver ->
+            [ group: grp, name: 'quasar-core-osgi', version: ver, ext: 'jar' ]
+        }
+        agent = dependency.map {
+            it['classifier'] = 'agent'
```

Comment:
doesn't this modify the above map
```groovy
[ group: grp, name: 'quasar-core-osgi', version: ver, ext: 'jar' ]
```
in place?

Reply:
`dependency` is a "lazy" Gradle `Provider`, and so is re-evaluated whenever you execute it. (Unless it has been finalized, which it isn't in this case.) Basically it's just a sequence of operations to generate a value.

Reply:
oh, ok... I thought the returned value was cached after the first execution and the closure never run again

Reply:
You _can_ make providers behave like that, if you want - see [finalizeValue](https://docs.gradle.org/current/javadoc/org/gradle/api/provider/HasConfigurableValue.html#finalizeValue--) and [finalizeValueOnRead](https://docs.gradle.org/current/javadoc/org/gradle/api/provider/HasConfigurableValue.html#finalizeValueOnRead--). _However_, Gradle may want to check these values several times and you wouldn't want to finalize them too early by accident.